src/pages/ProjectEditPage/hooks/useFileUpload.ts

```diff
@@ -0,0 +1,89 @@
+import { useEffect, useRef, useState } from "react"
+
+import { postProjectFiles } from "@api/projectFile/postProjectFiles"
+
+import { FileUploadStateType } from "../types/FileUploadStateType"
+
+export const useFileUpload = () => {
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const [files, setFiles] = useState<
+    FileUploadStateType<{ fileUrl: string }>[]
+  >([])
+
+  const onChangeFile = async (e: Event, fileLength: number) => {
+    if (!(e.target as HTMLInputElement).files) return
+    const selectFiles = (e.target as HTMLInputElement).files as FileList
+    const uploadFiles = Array.from(selectFiles)
+
+    if (files.length + uploadFiles.length > 6) {
+      alert("더이상 추가할 수 없습니다")
+      return
+    }
+
+    const promises = Object.values(uploadFiles).map((file, idx) => {
```

Comment:
```suggestion
    const promises = uploadFiles.map((file, idx) => {
```
[P3] : uploadFiles가 이미 배열이기 때문에 이렇게 변경해도 좋을 것 같습니다~

Reply:
이 부분은.. 저도 둘 다 동일한 배열이라고 생각하긴 하는데, 참고했던 자료에서 이런 방식으로 하더라구요.. 파일이라는 객체가 좀 특수한 부분도 있고 해서 한번 테스트해보고 수정하겠습니다